### DISCUSSION TITLE
Shopify Starter don't update the cart because process.env.GATSBY_STOREFRONT_ACCESS_TOKEN became undefined

### DISCUSSION THREAD

**Discussion by knupel** [gatsbyjs/gatsby#32938] [Help]
Hello, 
I install the Gatsby-Shopify starter. Every thing work like as expected, except the cart who stay in the `updating` status and stay empty, same result in local or when it's deployed on Gatsby cloud https://cafe366stan.gatsbyjs.io/

For information, the website Shopify is deployed officially with Shopify https://cafe-366.myshopify.com/ the problem can come from that ?
I have a feeling to make every thing like is necessary, or it's something I must do in my private app in Shopify setting ?

I've no idea to solve this problem.

![updating_cart_but_nothing_happen](https://user-images.githubusercontent.com/3332269/131100562-e91be527-6589-4cfb-aa58-0b397f0a9db3.jpg)
![shopify_setting](https://user-images.githubusercontent.com/3332269/131100998-46aa0e33-6291-4478-9c04-971cf7771fa0.jpg)

**Comment by knupel** [comment]
In addition: I see in the video https://youtu.be/58OD42jyF4o?t=1986 @LekoArts talk about this problem with the `localStorage` but i don't understand how to fix it ???? Because when I try to add something to cart from a different computers or phones, the same problem happen. So I'm not sure in my case it's a problem of `localStorage` Plus when I try https://shopify-demo.gatsbyjs.com/ I have no problem to add to cart.

**Comment by LekoArts** [comment]
Make sure that you also define `GATSBY_STOREFRONT_ACCESS_TOKEN` as an environment variable

**Comment by knupel** [comment]
@LekoArts ok thx. I don't add it before. Because in the template there is nothing written about the access token, I believe. So I create the authorization in Shopify for the `StoreFront` and add the info in `Gatsby` code, but that's don't work. :(  I make something like that, may be I miss something again !!!
```
GATSBY_STOREFRONT_ACCES_TOKEN="xxx"
GATSBY_SHOPIFY_STORE_URL="cafe-366.myshopify.com"
SHOPIFY_API_KEY="xxx"
SHOPIFY_SHOP_PASSWORD="shppa_xxx"

```
and in the config
```
{
      resolve: "gatsby-source-shopify",
      options: {
        password: process.env.SHOPIFY_SHOP_PASSWORD,
        storeUrl: process.env.GATSBY_SHOPIFY_STORE_URL,
        // accessToken: process.env.GATSBY_SHOPIFY_ACCESS_TOKEN,
        accessToken: process.env.GATSBY_STOREFRONT_ACCESS_TOKEN,
        shopifyConnections: ["collections"],
      },
    },
```

![acces_token](https://user-images.githubusercontent.com/3332269/131144220-e53649f9-2999-44b7-b609-dcd1c30eb9f0.jpg)

**Comment by knupel** [comment]
@LekoArts I find something strange with `acces token` when this one is call in the file [`src/context/store-context.jsx`](https://github.com/gatsbyjs/gatsby-starter-shopify/blob/main/src/context/store-context.jsx) is appearing like `undefined` To investigate on my problem I add a `console.log()`

```js
console.log(
  "process.env.GATSBY_STOREFRONT_ACCESS_TOKEN",
  process.env.GATSBY_STOREFRONT_ACCESS_TOKEN
);
console.log(
  "process.env.GATSBY_SHOPIFY_STORE_URL",
  process.env.GATSBY_SHOPIFY_STORE_URL
);

console.log("process.env.SHOPIFY_API_KEY", process.env.SHOPIFY_API_KEY);

console.log(
  "process.env.SHOPIFY_SHOP_PASSWORD",
  process.env.SHOPIFY_SHOP_PASSWORD
);
```
the console return 
```
process.env.GATSBY_STOREFRONT_ACCESS_TOKEN undefined
process.env.GATSBY_SHOPIFY_STORE_URL cafe-366.myshopify.com
process.env.SHOPIFY_API_KEY undefined
process.env.SHOPIFY_SHOP_PASSWORD undefined
```

And when I pass the `token` in hard-code that's work perfectly. May be it's normal than data became 'undefined` to protect the secret, but in this case how Gatsby can acces to the token ?

I've tried to create a file to save the `.env`but I failed to give this dotenv in memory https://dev.to/sanfra1407/how-to-use-env-file-in-javascript-applications-with-webpack-18df.  Sorry to disturbing with that, but i'm not an expert or pro in Javascript.
